core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java

```diff
@@ -141,9 +141,13 @@ protected void registerThroughputGauge(String counterName, String throughputName
         new Gauge<Object>() {
           @Override
           public Object getValue() {
-            long uptime = (System.currentTimeMillis() - mMetricsStore.getLastClearTime())
-                / Constants.SECOND_MS;
-            return uptime == 0 ? 0 : MetricsSystem.counter(counterName).getCount() / uptime;
+            // Divide into two lines so uptime is always zero or positive
+            long lastClearTime = mMetricsStore.getLastClearTime();
+            long uptime = (mClock.millis() - lastClearTime)
```

Comment:
I'm not sure if this will still fix the issue?

A java `Clock` object can have an offset added to it (See https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html). If the duration is negative it could cause this to return negative for an extended period of time.

I don't think we use the `offset` method anywhere, but it would be good to guard against it. I think keeping the code on one line it fine.

Maybe instead we can simply update the `return uptime == 0 ? ` to `return uptime <= 0 ? value : value / uptime`

or actually - since we should simply return the value within the first minute, maybe it should be 

```java
return uptime <= 60 * Constants.SECOND_MS ? value : value / uptime
```

Reply:
Clock.offset can be set, but usually it's in tests only. 
Changed to set uptime <= 0